catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplicationTest.java

```diff
@@ -0,0 +1,438 @@
+      Source xslt = new StreamSource(getResourceFile("inline-xml.xslt"));
+      Transformer transformer = factory.newTransformer(xslt);
+
+      Source text = new StreamSource(new ByteArrayInputStream(xml.getBytes()));
+      StringWriter destination = new StringWriter();
+
+      transformer.transform(text, new StreamResult(destination));
+
+      // Replace the header that the XSLT transform removed
+      return destination
+          .toString()
+          .replace(
+              "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+              "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>");
+    } catch (TransformerException e) {
+      throw new AssertionError("Could not remove pretty printing from XML, " + xml, e);
+    }
+  }
+
+  private static FilterWriter getWriter() {
+    try {
+      return new FilterWriter(true);
+    } catch (JAXBException e) {
+      throw new AssertionError("Could not make filter writer, " + e.getMessage());
+    }
+  }
+
+  private static AttributeRegistry getRegistry() {
+    AttributeRegistry registry = mock(AttributeRegistry.class);
+    doReturn(Optional.empty()).when(registry).lookup(any());
+    return registry;
+  }
+
+  private static File getResourceFile(String... resourceRoute) {
+    try {
+      File dir = new File(FILTER_RESOURCES_DIR.toURI());
+      if (!dir.exists()) {
+        fail(
+            format(
+                "Invalid setup parameter '%s', the directory does not exist",
+                FILTER_RESOURCES_DIR.toString()));
+      }
+
+      Path route = Arrays.stream(resourceRoute).map(Paths::get).reduce(Path::resolve).orElse(null);
+      if (route == null) {
+        fail("Could not reduce resource route to a single path");
+      }
+
+      File resourceFile = dir.toPath().resolve(route).toFile();
+      if (!resourceFile.exists()) {
+        fail("File was not found " + resourceFile.getAbsolutePath());
+      }
+
+      return resourceFile;
+    } catch (URISyntaxException e) {
+      throw new AssertionError("Could not complete test setup due to exception", e);
+    }
+  }
+
+  private static String getContentsOfFile(String... resourceRoute) {
+    File resourceFile = getResourceFile(resourceRoute);
+    try (FileInputStream fis = new FileInputStream(resourceFile)) {
+      return IOUtils.toString(fis, StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      throw new AssertionError("Could not complete test setup due to exception", e);
+    }
+  }
+
+  /**
+   * Given a port range, this function returns the first port within the range that it's able to
+   * bind to. The range is inclusive on both sides.
+   *
+   * <p>The discovered port should be available for another client to bind to by the time this
+   * function has returned. Given this detail, running unit tests on environments that are in the
+   * process of being provisioned or are otherwise in a state of flux may cause erroneous failures
+   * due to port binding race conditions.
```

Comment:
ℹ️ Some background context is needed here. 

For running unit tests, or builds in general, it's not always a safe assumption that the environment wouldn't be changing. But in my mind it seems practical enough for most cases. Especially since:
1. Why are **tests** running when the host environment is not in a **controlled**, stable state? 
2. This is the only **unit** test interacting with `ServerSocket` (curious how else to evaluate if sockets are getting provisioned indirectly by unit tests; likely will require runtime monitoring)

Why was this solution implemented? Mostly because of Spark...

- Spark's javadoc claims that `Spark.port(0)` will trigger a bind to _some_ available port. This is correct and works fine, but `Spark.port()` called after the bind returns `0` which doesn't help me construct my URL for the test. This looks like a bug in Spark (https://github.com/perwendel/spark/blob/2.5.5/src/main/java/spark/Service.java#L452). 
- Looked into extending / overriding Spark facilities so I can get the port I need; it's non-trivial and likely not even possible. 
- Thought about just doing the upgrade to Javalin but that introduces a lot of risk since all of catalog-ui-search would, at this point, need to be upgraded. 
- Tried using Spark itself as the binding check but it does not throw any exceptions when it fails. 
- Found the `PortFinder` utility that handles port locking similar to the iTests but it won't solve the issue since the port would have to be unbound and given to Spark, which still leaves a window for the race condition.

Reply:
❗️ Update: revisited the docs and found the `initExceptionHandler` (http://sparkjava.com/documentation#stopping-the-server). Will try it to see if the race condition can be resolved.

Reply:
We are using Spark `2.5.5` and this requires at a minimum Spark `2.6.0`. This will not work without upgrading.